### PR TITLE
Unify expiration handling for SingleUseObjects

### LINKED ISSUE #12205: Unify expiration for SingleUseObjects
### Description

Follow up on #11947 for SingleUseObjects store

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### LINKED ISSUE #12563: Make expiration check automatic within each storage implementation
### Description

Currently, each `ModelCriteriaBuilder` for searching entities that can expire contains `EXPIRATION` `SearchableField` for filtering non-expired entities. Each storage implementation should handle not returning expired entities itself.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mhajas**
Closes #12205
Closes #12563

### CODE REVIEW

**Review by martin-kanis — COMMENTED** [review]
Thanks @mhajas. I have one question.

File: model/map/src/main/java/org/keycloak/models/map/common/ExpirationUtils.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * Checks whether the {@code entity} is expired
+     *
+     * @param entity to check
+     * @return true if the {@code entity} is expired (expiration time is in the past or now), false otherwise
+     */
+    public static boolean isExpired(ExpirableEntity entity) {
+        Long expiration = entity.getExpiration();
+        return expiration != null && expiration <= Time.currentTimeMillis();
```
[the remark this answers is not in the document]
Comment (mhajas) [reply model/map/src/main/java/org/keycloak/models/map/common/ExpirationUtils.java:32]: Good point. However, the behaviour you described conflicts with [Javadoc here](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/common/ExpirableEntity.java). We should discuss this, I am adding discussion point to the meeting today.

File: testsuite/model/src/test/java/org/keycloak/testsuite/model/KeycloakModelTest.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import org.keycloak.authorization.DefaultAuthorizationProviderFactory;
 import org.keycloak.authorization.store.StoreFactorySpi;
 import org.keycloak.cluster.ClusterSpi;
+import org.keycloak.common.util.Time;
```
Comment (hmlnarik) [inline testsuite/model/src/test/java/org/keycloak/testsuite/model/KeycloakModelTest.java:27]: unused import?

Comment (mhajas) [reply testsuite/model/src/test/java/org/keycloak/testsuite/model/KeycloakModelTest.java:27]: It seems I did some mistake during rebase. The changes in the model tests should not be there. I will fix this

File: model/map/src/main/java/org/keycloak/models/map/events/MapEventStoreProvider.java [hunk 3]
```diff
@@ -123,14 +123,6 @@ public void clear(RealmModel realm, long olderThan) {
     @Override
     public void clearExpiredEvents() {
         LOG.tracef("clearExpiredEvents()%s", getShortStackTrace());
```
[the remark this answers is not in the document]
Comment (mhajas) [reply model/map/src/main/java/org/keycloak/models/map/events/MapEventStoreProvider.java:125]: Warning added. The method is already deprecated.

[the remark this answers is not in the document]
Comment (hmlnarik) [reply model/map/src/main/java/org/keycloak/models/map/events/MapEventStoreProvider.java:125]: Excellent. Thank you!

**Review by hmlnarik — APPROVED** [review]
Thank you for the PR @mhajas! Great job!

**Pull request merged**